File: Lista0/Lista0/ciagPunktow3D.cpp
#include <iostream>
#include <list>
#include <algorithm>

#include "ciagPunktow3D.h"
#include "punkt3D.h"
#include <iterator>

using namespace std;

ciagPunktow3D::ciagPunktow3D(){
	list<punkt3D> l = {};
}

void ciagPunktow3D::dodajPunkt(const punkt3D p) {
	l.push_back(p);
}

void ciagPunktow3D::usunPunkt(const punkt3D p) {
    auto szukany = find(l.begin(), l.end(), p);

    if (szukany != l.end()) {
        l.erase(szukany);
    }
    else {
        cout << "Nie znaleziono punktu! ";
    }

}

bool ciagPunktow3D::czyPunktJestWCiagu(punkt3D p) {
    auto szukany = find(l.begin(), l.end(), p);
    return szukany != l.end();
}

punkt3D ciagPunktow3D::podajPunkt(int i) {
    if (i < 0 || i > l.size()) {
        throw out_of_range("Podano nieprawidlowy indeks! ");
    }
    auto it = l.begin();
    advance(it, i);
    return *it;
   
}

int ciagPunktow3D::dlugoscCiagu() {
    return l.size();
}

void ciagPunktow3D::wyswietl() const {
    cout << "[";
    for (const auto& p : l) {
            cout <<" (" << p.wezX() << ", " << p.wezY() << ", " << p.wezZ() << ")";
    }
    cout << " ]";
}

File: lista0java/src/main/java/com/mycompany/lista0java/CiagPunktow3D.java
package com.mycompany.lista0java;

import java.util.ArrayList;
import java.util.List;

public class CiagPunktow3D {

    private List<Punkt3D> lista;

    public CiagPunktow3D() {
        lista = new ArrayList<>();
    }

    public void dodajPunkt(Punkt3D p) {
        lista.add(p);
    }

    public void usunPunkt(Punkt3D p) {
        if (!lista.remove(p)) {
            System.out.println("Nie znaleziono punktu!");
        }
    }

    public boolean czyPunktJestWCiagu(Punkt3D p) {
        return lista.contains(p);
    }

    public Punkt3D podajPunkt(int i) {
        if (i < 0 || i >= lista.size()) {
            throw new IndexOutOfBoundsException("Podano nieprawidlowy indeks!");
        }
        return lista.get(i);
    }

    public int dlugoscCiagu() {
        return lista.size();
    }

    public void wyswietl() {
        System.out.print("[ ");
        for (Punkt3D p : lista) {
            System.out.print(p + " ");
        }
        System.out.println("]");
    }
}